app/src/main/kotlin/com/github/gotify/MarkwonFactory.kt

```diff
@@ -68,6 +73,19 @@ internal object MarkwonFactory {
                     imageLoader
                 )
             )
+            .usePlugin(object : AbstractMarkwonPlugin() {
+                override fun configureSpansFactory(builder: MarkwonSpansFactory.Builder) {
+                    builder.appendFactory(Image::class.java) { configuration, props ->
```

Comment:
The alternative to opening the browser would be a library with an image viewer that supports zoom etc. This image view then could reuse the already cached image, so it doesn't have to be downloaded again. Tho, this benefit is probably negligible. 

@cyb3rko Do you have an opinion here? You've mentioned in the ticket https://github.com/gotify/android/issues/348 that you have a look at it.

Reply:
Personally I would prefer an image view with that functionality instead of opening it externally.  
But I didn't have a look at it yet

Reply:
@Ghost-LZW I agree with this, I rather have some kind of native image viewer instead of opening this in the browser. This likely also fixes my first comment.

Reply:
One problem would be zooming because the image view is part of a recyclerview item.  
Therefore I'm currently testing it with an image view that opens the image in fullscreen when you click it.  
In that full screen you can zoom and move around in the image.

Does that sound like a solution?

Original: https://github.com/stfalcon-studio/StfalconImageViewer  
Updated fork: https://github.com/cyb3rko/fullscreen-image-viewer

Reply:
I tried to use such as https://github.com/panpf/zoomimage to display the image. But I'm not an Android expert, It seems like it will take more time than I expected. I'll close this PR and wait for your progress.

Reply:
That library looks nice, I will check it out.  
Maybe a bit overpowered for our use case.

Reply:
Yeah, but at least it's maintained the other one hasn't received an update in 3 years.